src/Template/Snippet.php

```diff
@@ -128,6 +128,57 @@ public static function end(): void
 		echo static::$current?->render();
 	}
 
+	/**
+	 * Loads the code of the given snippet.
+	 *
+	 * @param string $file Path to the snippet file
+	 * @param array $data Data available inside the snippet
+	 * @param \Kirby\Template\Slots|null Slots available in the snippet
+	 * @return string The rendered content of the given snippet file
+	 */
+	public static function load(
+		string|null $file = null,
+		array $data = [],
+		Slots|null $slots = null
+	): string {
+		// if the snippet is rendered inside another snippet,
+		// we need to keep the "outside" snippet to compare it later
+		$outside = Snippet::$current;
+
+		// load the snippet code
+		$snippet = parent::load(
+			file: $file,
+			data: static::scope($data, $slots)
+		);
+
+		// if last `endsnippet()` inside the current template
+		// has been omitted (= snippet was used as layout snippet),
+		// `Snippet::$current` will point to a snippet that was
+		// opened inside the template; if that snippet is the direct
+		// child of the snippet that was open before this snippet was
+		// rendered (which could be `null` if no snippet was open),
+		// take the buffer output from the template as default slot
+		// and render that nested snippet as final snippet output
+		if (
+			Snippet::$current === null ||
+			Snippet::$current->parent() !== $outside
+		) {
+			return $snippet;
+		}
+
+		if (Snippet::$current->slots()->count() === 0) {
+			// no slots have been defined, but the snippet code
+			// should be used as default slot
+			return Snippet::$current->render($data, [
+				'default' => $snippet
+			]);
+		}
+
+		// swallow any "unslotted" content
+		// between start and end
+		return Snippet::$current->render($data);
```

Comment:
`Help:` I cannot find a unit test to cover this line. I am wondering if there is a logical mistake and this line is never reachable.

Reply:
I added unit tests for it and they run through, but the line is still not marked as covered. If I remove the line from the class, my two new tests fail though. I'm pretty sure this is a Codecov issue.